plugins/woocommerce/includes/class-wc-frontend-scripts.php

```diff
@@ -45,10 +45,38 @@ class WC_Frontend_Scripts {
 	 */
 	public static function init() {
 		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'load_scripts' ) );
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'warn_deprecated_scripts' ) );
 		add_action( 'wp_print_scripts', array( __CLASS__, 'localize_printed_scripts' ), 5 );
 		add_action( 'wp_print_footer_scripts', array( __CLASS__, 'localize_printed_scripts' ), 5 );
 	}
 
+	/**
+	 * Log a warning for deprecated scripts/styles that will be removed.
+	 */
+	public static function warn_deprecated_scripts() {
+		global $wp_scripts, $wp_styles;
+
+		if ( wp_script_is( 'prettyPhoto', 'enqueued' ) ) {
+			// phpcs:ignore
+			error_log( "The 'prettyPhoto' script has been deprecated and will be removed in a future version." );
+		}
+
+		if ( wp_script_is( 'prettyPhoto-init', 'enqueued' ) ) {
+			// phpcs:ignore
+			error_log( "The 'prettyPhoto-init' script has been deprecated and will be removed in a future version." );
```

Comment:
Perhaps we can add another function in `wc-deprecated-functions.php` which throws an error when `WP_DEBUG` is enabled and the request is not ajax/api, just like other `wc_deprecated_*` series of functions. wdyt?

Reply:
@vedanshujain instead of this approach or in addition to this approach?

Reply:
I'd say addition to, but let the function in wc-deprecated file take care of whether to log or throw an error.